Generation/src/components/HepMCSimpleMerge.cpp

```diff
@@ -0,0 +1,59 @@
+
+#include "HepMCSimpleMerge.h"
+
+#include "GaudiKernel/IIncidentSvc.h"
+#include "GaudiKernel/Incident.h"
+#include "GaudiKernel/IEventProcessor.h"
+
+#include "GaudiKernel/DeclareFactoryEntries.h"
+
+DECLARE_TOOL_FACTORY(HepMCSimpleMerge)
+
+HepMCSimpleMerge::HepMCSimpleMerge( 
+    const std::string& type,
+    const std::string& name, 
+    const IInterface* parent)
+    : GaudiTool(type, name, parent) {
+  declareInterface< IHepMCMergeTool >( this );
+}
+
+HepMCSimpleMerge::~HepMCSimpleMerge() {
+}
+
+StatusCode HepMCSimpleMerge::initialize() {
+  StatusCode sc = GaudiTool::initialize();
+  if (!sc.isSuccess()) return sc;
+  return sc;
+}
+
+HepMC::GenEvent* HepMCSimpleMerge::merge(const std::vector<HepMC::GenEvent>& eventVector) {
+  // initial event is assumed to be signal event -- copied in full
+  HepMC::GenEvent* pileEvent = new HepMC::GenEvent(eventVector[0]);
```

Comment:
Since you delete at the end, this should either be a smart pointer or no pointer at all.